Tests/BSONTests/CodecTests.swift

```diff
@@ -41,6 +41,14 @@ final class CodecTests: BSONTestCase {
         let s: NestedStruct
     }
 
+    /// Test decoding non-document BSON
+    func testAnyBSON() throws {
```

Comment:
could we add a case here that uses one of the types introduced in the library (e.g. `ObjectId`)? This will verify whether the changes above were adequate.

Reply:
added